viewing_party/party.py

```diff
@@ -1,23 +1,360 @@
+        - add that movie to watched
+        - return the `user_data`
+    - If the title is not a movie in the user's watchlist:
+        - return the `user_data`
+    '''
+    # using the for loop to iterate through the list user_data["watchlist"]
+    for movie in user_data["watchlist"]:
+        # and check if the movie's title is what we're looking for,    
+        if movie["title"] == title:
+            # then we add it to the "watched"            
+            user_data["watched"].append(movie)
+            # and remove it from the "watchlist"           
+            user_data["watchlist"].remove(movie)           
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    '''
+    - take one parameter: `user_data`
+    - Calculate the average rating of all movies in the watched list
+    - The average rating of an empty watched list is `0.0`
+    - return the average rating
+    '''
+    #creating basic variables that we'll need
+    average_rating = 0.0
+    total = 0.0
+
+    #if the "watched is empty, we return 0.0"
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    else:
+        #using the for loop to iterate through every movie in "watched"
+        for movie in user_data["watched"]:
+            #and calculate total sum of movie ratings
+            total += movie["rating"]
+        #number of movies equals the length of "watched"    
+        movie_count = len(user_data["watched"])
+        average_rating = total/movie_count
+        return average_rating    
+
+def get_most_watched_genre(user_data):
+    '''
+    - take one parameter: `user_data`
+    - Determine which genre is most frequently occurring in the watched list
+    - return the genre that is the most frequently watched
+    - If the value of "watched" is an empty list, `get_most_watched_genre` should return `None`.
+    
+    '''
+
+    # creating an empty dictionary fo genres to calculate frequency
+    # genre will be a key and frequency a value
+    genres = {} 
+    if len(user_data["watched"]) == 0: #if "watched" is empty, we return None
+        return None
+    else:
+        #using a for loop to iterate through "watched"
+        for movie in user_data["watched"]:
+            #if it's a first time we encounter this genre we create key/value pair with value "1"
+            if movie["genre"] not in genres:
+                genres[movie["genre"]] = 1
+            #if we already have this key, we add 1 to the value
+            else:
+                genres[movie["genre"]] += 1
+        
+        #Here I wanted to use max function and found a method to get the specific key corresponding to this max value.
+        #I'm not checking an edge case where there's two exact same frequencies with maximum value
+        most_watched_genre = max(genres, key=genres.get)

```

Comment:
Great use of the max function, and a really great job on this algorithm overall!